File: pset6/credit.py
from cs50 import get_int
import sys

number = -1

while number < 0:
    number = get_int("Number: ")

digits = len(str(number))


if digits != 13 and digits != 15 and digits != 16:
    print("INVALID")
    sys.exit()

card = str(number)
sum1 = 0
product = 0

if digits % 2 == 0:
    for i in range(digits):
        num = int(card[i])
        if i % 2 == 0:
            product = num * 2
            if product > 9:
                sum1 += product // 10
                sum1 += product % 10
            else:
                sum1 += product
        else:
            sum1 += num
else:
    for i in range(digits):
        num = int(card[i])
        if i % 2 == 0:
            sum1 += num
        else:
            product = num * 2
            if product > 9:
                sum1 += product // 10
                sum1 += product % 10
            else:
                sum1 += product

if sum1 % 10 != 0:
    print("INVALID")
    sys.exit()
    
if int(card[0]) == 5 and 0 < int(card[1]) < 6:
    print("MASTERCARD")
elif int(card[0]) == 3 and (int(card[1]) == 4 or int(card[1]) == 7):
    print("AMEX")
elif int(card[0]) == 4:
    print("VISA")
else:
    print("INVALID")

File: pset1/credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    long number;
    long digits;
    int count = 0;
    
    do
    {
        number = get_long("Number: ");
    }
    while (number < 0);
    
    digits = number;
    
    while (digits > 0)
    {
        digits = digits / 10;
        count++;
    }
    if (count != 13 && count != 15 && count != 16)
    {
        printf("INVALID\n");
        return 0;
    }
    
    long card = number;
    int sum = 0;
    int product = 0;
    
    do
    {
        if (count == 16 || count == 13 || count == 15)
        {
            sum += card % 10;
            card = card / 10;
            product += card % 10;
            product += card % 10;
            sum += product % 10;
            product = product / 10;
            sum += product % 10;
            card = card / 10;
            product = 0;
        }
        
    }
    while (card > 0);
         
    if (sum % 10 != 0)
    {
        printf("INVALID\n");
        return 0;
    }
    
   
    
    long f2d = number;
    do
    {
        f2d = f2d / 10;
    }
    while (f2d > 100);
    
    if ((f2d / 10 == 5) && (0 < f2d % 10 && f2d % 10 < 6))
    {
        printf("MASTERCARD\n");
    }
    else if ((f2d / 10 == 3) && (f2d % 10 == 4 || f2d % 10 == 7))
    {
        printf("AMEX\n");
    }
    else if (f2d / 10 == 4)
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }
    
}